File: src/main/org/nlogo/api/JobOwner.java
package org.nlogo.api;

public interface JobOwner
    extends SourceOwner {
  String displayName();

  boolean isButton();

  boolean isTurtleForeverButton();

  boolean isLinkForeverButton();

  boolean ownsPrimaryJobs();

  boolean isCommandCenter();

  org.nlogo.util.MersenneTwisterFast random();
}

File: src/main/org/nlogo/api/JobOwner.scala
package org.nlogo.api

import org.nlogo.util.MersenneTwisterFast 

trait JobOwner extends SourceOwner {
  def displayName: String
  def isButton: Boolean
  def isTurtleForeverButton: Boolean
  def isLinkForeverButton: Boolean
  def ownsPrimaryJobs: Boolean
  def isCommandCenter: Boolean
  def random: MersenneTwisterFast
}
